crm_opportunity_currency/models/crm_lead.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2018 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from odoo import api, fields, models
+
+
+class CrmLead(models.Model):
+    _inherit = 'crm.lead'
+
+    customer_currency_id = fields.Many2one(
+        string='Customer currency',
+        comodel_name='res.currency',
+        default=lambda self: self.env.user.company_id.currency_id,
+    )
+    amount_customer_currency = fields.Monetary(
+        string='Customer amount',
+        currency_field='customer_currency_id',
+        store=True,
+    )
+    is_same_currency = fields.Boolean(
+        string='Same currency',
+        compute='_compute_is_same_currency',
+    )
+
+    @api.onchange('customer_currency_id', 'amount_customer_currency')
+    def _onchange_currency(self):
+        for lead in self:
```

Comment:
In the onchange method is not necessary to make for, this is not multi.

[Here](https://www.odoo.com/documentation/10.0/reference/orm.html#odoo.api.onchange) the Odoo doc

Reply:
Thanks changed.